feat(github)!: remove forkMode setting (#18074)
renovatebot/renovate: docs/usage/configuration-options.md

<!-- prettier-ignore -->
!!! note

>     The `baseBranches` config option is not supported when a `forkToken` has been configured, such as in the Forking Renovate app.

Reviewer: ```suggestion
    Do _not_ use the `baseBranches` config option when you've set a `forkToken`.
    You may need a `forkToken` when you're using the Forking Renovate app.
```

I think we should put the warning up front, and then explain _why_.
